### DISCUSSION TITLE
Child diagram graph of glsp editor

### DISCUSSION THREAD

**Discussion by mishaltripathi** [eclipse-glsp/glsp#995] [Q&A] [answered]
Hello All,

Greetings for the day!
We are developing an application based on theia and glsp framework. Our requirement is to have child diagram of glsp editor or can have multiple glsp editor of same diagram type. So would be wondering how currently its supported. Since we are seeing incompatibility with theia glsp integration plugins.

Any leads would be appreciated.

Thank you
Mishal

**Comment by planger** [comment]
Hi Mishal,
can you please specify in more detail what you mean by child diagrams or multiple glsp editors of the same diagram type?
Do you mean multiple diagram widgets within one editor widget, that is, one editor is showing multiple diagrams of the same type?

If yes, showing multiple diagram widgets within one editor isn't something that I've personally tried yet. But I guess it should work, if you create a [custom Theia widget](https://theia-ide.org/docs/widgets/) and add multiple [GLSP diagram widgets](https://github.com/eclipse-glsp/glsp-theia-integration/blob/master/packages/theia-integration/src/browser/diagram/glsp-diagram-widget.ts#L93) as its children. Widgets can contain other widgets in Theia so that should work as long as you provide all the information the child widgets need to get initialized properly via DI.
But that likely requires some customization of certain parts... best have a look at the code of the [Theia integration](https://github.com/eclipse-glsp/glsp-theia-integration/tree/master/packages/theia-integration/src/browser) itself to see how the GLSP Theia widget is initialized and replicate that for your custom widget and its child GLSP widgets.

Hope that gives some pointers into the right direction!

**Reply by mishaltripathi** [reply]
Thanks Philip for your valuable input on this. However I am trying to solve a problem where we have one model file along with multiple diagram files as below structure.
**Workspace folder**
-- **sample.coffee**
-- **sample1.notation**
-- **sample2.notaiton**
-- **sample3.notation**

These notation(diagram) files can be n number and should able to load the into diagram editor. Wondering if its feasible to implement?

Thanks
Mishal

**Reply by planger** [reply, marked as answer]
I see, yes I don't see any blockers to implement that. Your source model loader would need to be able to load all of these notation file and depending on some custom parameter or URI decide for which to create the GModel.

**Reply by mishaltripathi** [reply]
Thanks a lot Philip for your input.
